Devel logging

dudoslav opened the pull request:
Implementation of logging for cloudkeeper-aws.

dudoslav committed:
Add logging setup for cloudkeeper-aws

dudoslav committed:
Add logging into core classes

dudoslav committed:
Fix rubocop issues

dudoslav committed:
Fix specs and typo

Change constant name SUFFIX to PREFIX.

amarthadan commented on lib/cloudkeeper/aws/cloud.rb:
```diff
@@ -29,6 +29,7 @@ def initialize(s3service: nil, ec2service: nil)
       # @yield [write_stream] output stream
       # @raise [Cloudkeeper::Aws::Errors::BackendError] if file already exists
       def upload_data(file_name, &block)
+        logger.debug { "Uploading file: #{file_name}" }
```
Please also add to which bucket is image being uploaded. The same go for the `delete_data` method.

amarthadan commented on lib/cloudkeeper/aws/cloud.rb:
```diff
@@ -49,6 +51,7 @@ def delete_data(file_name)
       # @param appliance [Appliance] data about image
       # @return [Number] import task id
       def start_import_image(appliance)
+        logger.debug { "Starting import image task for #{appliance}" }
```
I think this will print useless object description for appliance instead of its identifier.

amarthadan commented on lib/cloudkeeper/aws/cloud.rb:
```diff
@@ -111,9 +121,8 @@ def timeout
-        ec2.deregister_image(
-          image_id: image_id
-        )
+        logger.debug { "Deregistering image ##{image_id}" }
```
What is `image_id` in this context? An AMI's ID? If so, please use it in the log to distinguish between bucket image and AMI image. Same goes for all AMI specific operations and their logging.

amarthadan reviewed (changes requested):
Fix it!

dudoslav committed:
Rework logging to be more specific

Logging now distinguishes between images (cloudkeeper images) and AMIs (AWS images)

dudoslav committed:
Add config logging during application startup

amarthadan reviewed (approved):
:avocado:
